### PR TITLE
Migrating to TypeScript 3.7.2

### PR DESCRIPTION
**Pull request by MaxFlower**
- migrating js to ts
- turn js and css files into one bundle file
- refactor the functions set of arguments
- commonJs importing support
- styling in accordance with company design color scheme
- migrate tests
- send additional property in person object

### CODE REVIEW

**Comment by MaxFlower** [comment]
TBD:
- commonJs import support
- migrate tests
- send additional property in person object

File: src/template.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  lastName: string | boolean;
+  idNumber: string | boolean;
+  vendorData: string | boolean;
+}
+
+const defaultFormLabel: IFormLabel = {
+  givenName: 'Given name',
+  lastName: 'Last name',
+  idNumber: 'Id number',
+  vendorData: 'Vendor data',
```
Comment (suur) [inline src/template.ts]: "vendor" is a deprecated term we shouldn't use publicly

File: src/template.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  givenName: string;
+  lastName: string;
+  idNumber: string;
+  vendorData: string;
+}
+interface IPerson {
+  givenName: string | boolean;
+  lastName: string | boolean;
+  idNumber: string | boolean;
+  vendorData: string | boolean;
```
Comment (suur) [inline src/template.ts]: it would make sense to refactor this into something more concrete

File: src/util.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export function capitalize(str) {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+export function camelCaseToSlug(str) {
+  return str.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase();
+}
+
+export function camelCaseHuminize(str) {
```
Comment (suur) [inline src/util.ts]: Humanize :)

File: src/xhr.ts [hunk 4]
```diff
@@ -1,37 +1,39 @@
-const createSession = function(host = 'https://api.veriff.me', apiKey, data, cb) {
+export function createSession(host, apiKey, data, cb) {
   const url = `${host}/v1/sessions`;
-  const xhr = new XMLHttpRequest();
+  const xhr: XMLHttpRequest = new XMLHttpRequest();
```
Comment (suur) [inline src/xhr.ts]: the type definition here seems redundant

File: src/styles/style.css [hunk 5]
```diff
@@ -1,39 +1,76 @@
...
+  width: 100%;
+  background-color: #004e5f;
+  color: white;
+  text-transform: uppercase;
+  font-weight: 500;
+  font-size: 16px;
+  line-height: 40px;
+  letter-spacing: 0.4px;
+
+  display: inline-flex;
```
Comment (suur) [inline src/styles/style.css]: we are targeting ES5 but using flexbox. are there ES5 browsers with flex support?

File: src/template.ts [hunk 6]
```diff
@@ -0,0 +1,123 @@
+import { camelCaseToSlug } from './util';
+
+interface IFormLabel {
```
Comment (janari) [inline src/template.ts]: Look like its identical to `IPerson `.

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  const { host = 'https://api.veriff.me', apiKey, parentId, onSession } = options;
+  return {
+    params: {
+      person: {},
+    },
+    setParams(newParams: IParams) {
+      this.params = { ...this.params, ...newParams };
+    },
+    updateParams(newParams: IParams) {
+      this.params = { ...newParams };
```
Comment (suur) [inline src/index.ts]: setParams "patches" the params with new values, but updateParams replaces them. is this intentional?

File: src/index.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      this.form = this.assignSubmit(this.form);
+    },
+    form: this.form,
+    assignSubmit(form, loadingText = 'Loading...', submitBtnText = 'Start Verification') {
+      form.onsubmit = (e) => {
+        e.preventDefault();
+
+        const givenName = form.givenName ? form.givenName.value : this.params.person.givenName;
+        const lastName = form.lastName ? form.lastName.value : this.params.person.lastName;
+        const idNumber = form.idNumber ? form.idNumber.value : this.params.person.idNumber;
```
Comment (suur) [inline src/index.ts]: are we keeping the id number in the JS SDK? talk to Maksim, maybe we can simply drop it. we'll keep existing integrations intact by versioning all new releases

File: src/xhr.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  host: string,
+  apiKey: string,
+  data: { person?: IPersonData; vendorData?: string },
+  cb: (statusObject, resp) => void
+): void {
+  const url = `${host}/v1/sessions`;
+  const xhr = new XMLHttpRequest();
+  xhr.open('POST', url, true);
+  xhr.setRequestHeader('Content-type', 'application/json');
+  xhr.setRequestHeader('x-auth-client', apiKey);
```
Comment (suur) [inline src/xhr.ts]: one more thing. let's set 'x-origin': 'js-sdk' to better analyze session sources

**Pull request merged**